### ISSUE TITLE
Node: Update "Forms and Data Handling" lesson to use a single GET request for search functionality

### ISSUE THREAD

**Issue by cakegod** [TheOdinProject/curriculum#28585]
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### Describe your suggestion

The current assignment in the "Forms and Data Handling" lesson suggests implementing a `/search` route that accepts both GET and POST requests:
![image](https://github.com/user-attachments/assets/2a3ac12f-485a-4e23-90cc-6641d1c306e8)
This approach is not ideal for a search functionality, as searching is a data retrieval operation, it is idempotent and does not contain sensitive data. Therefore, this functionality should be implemented with a single GET request.

The GET request can be implemented with a form GET method. The data will be passed as a query string. We can then access this data in express js through `req.query`.

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/nodejs-forms-and-data-handling

### (Optional) Discord Name

Cake

**Comment by MaoShizhong** [comment]
Thanks for the feedback and suggestion - agreed that a search form should be GET only (which is exactly what's said in the lesson as well). On top of that, I think the last paragraph of [HTML forms overview](https://www.theodinproject.com/lessons/nodejs-forms-and-data-handling#html-forms-overview) can clarify that POST puts form data in the request body, and GET forms put the form data in the URL, which will be parsed as a query string. e.g:

> `POST` is generally more secure because it keeps sensitive information out of the URL by putting the form data in the request body, which means they won’t show up in server logs, and is the standard choice for creating or updating data on the server side. `GET` is for forms that don’t modify data, such as search forms, or when you want the form submission to be bookmarkable or shareable via URL. The form data here is sent as a query string as part of the request URL.

## Acceptance criteria

- [ ] The [Implement searching](https://www.theodinproject.com/lessons/nodejs-forms-and-data-handling#implement-searching) assignment is amended to:
  - [ ] specify the search form should be a `GET` form
  - [ ] instruct to create a `/search` route for the `GET` method
  - [ ] parse the search query via `req.query` and render accordingly
- [ ] The last paragraph of [HTML forms overview](https://www.theodinproject.com/lessons/nodejs-forms-and-data-handling#html-forms-overview) is amended to the above example, or something similar.

Please comment below if you'd like to be assigned to this issue. Please do not open a PR unless you have been assigned by a maintainer.

**Comment by DrantDumani** [comment]
I would like to be assigned this issue.
